src/v/cloud_storage/types.cc

```diff
@@ -252,6 +252,16 @@ bool anomalies::has_value() const {
            || segment_metadata_anomalies.size() > 0;
 }
 
+size_t anomalies::count_segment_meta_anomaly_type(anomaly_type type) const {
+    auto begin = segment_metadata_anomalies.begin();
+    auto end = segment_metadata_anomalies.end();
+    const auto count = std::count_if(begin, end, [&type](const auto& anomaly) {
```

Comment:
Is it? `segment_metadata_anomalies` may contain different types of anomalies. The function counts anomalies of a specific type.

Reply:
```cpp
enum class typ {a, b};

std::vector<typ> anomalies{typ::a, typ::a, typ::b};
std::cout << std::count(anomalies.begin(), anomalies.end(), typ::a); // 2
```

[godbolt](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:13,positionColumn:2,positionLineNumber:13,selectionStartColumn:2,selectionStartLineNumber:13,startColumn:2,startLineNumber:13),source:'%23include+%3Calgorithm%3E%0A%23include+%3Ciostream%3E%0A%23include+%3Cvector%3E%0A%0Aenum+class+typ+%7B%0A++++a,%0A++++b,%0A%7D%3B%0A%0Aint+main()+%7B%0A++++std::vector%3Ctyp%3E+anomalies%7Btyp::a,+typ::a,+typ::b%7D%3B%0A++++std::cout+%3C%3C+std::count(anomalies.begin(),+anomalies.end(),+typ::a)%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:66.0449437885826,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:output,i:(compilerName:'x86-64+gcc+13.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+clang+(trunk)+(Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:clang_trunk,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+clang+(trunk)+(Editor+%231)',t:'0')),k:33.95505621141739,l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:33.95505621141739,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

Reply:
Right, but applied to the code in question it would be more like since the objects returned by the iter
are not of type `anomaly_meta`, not `anomaly_type`.
```
enum class typ {
    a,
    b,
};

struct data {
    typ type;
    int value;
};

std::vector<data> datas{data{.type = typ::a, .value = 1}, {.type = typ::b, .value = 1}};
std::cout << std::count(datas.begin(), datas.end(), typ::a);
```

Maybe I'm missing something though.

Reply:
My bad 🤦‍♂️. But, `std::cout << std::ranges::count(datas, typ::a, &data::type);`